graphql/resolvers/_queries/statements.js

```diff
@@ -0,0 +1,108 @@
+const isUUID = require('is-uuid')
+const {
+  transformUser
+} = require('@orbiting/backend-modules-auth')
+
+module.exports = async (_, args, { pgdb, t }) => {
+  const {first, after, search, focus} = args
+  const seed = args.seed || Math.random() * 2 - 1
+
+  if (first > 100) {
+    throw new Error(t('api/statements/maxNodes'))
+  }
+
+  let firstId
+  if (focus) {
+    firstId = await pgdb.public.users.findOneFieldOnly({
+      or: isUUID.v4(focus)
+        ? [
+          {testimonialId: focus},
+          {id: focus}
+        ]
+        : [{slug: focus}]
+    }, 'id')
+  }
+
+  const results = async (userRows) => {
+    let ids = userRows.map(user => user.id)
+    if (firstId) {
+      ids = [firstId].concat(ids)
+    }
+
+    let startIndex = 0
+    let endIndex = ids.length
+    if (after) {
+      startIndex = ids.findIndex(id => id === after)
+    }
+    endIndex = startIndex + first
+
+    const nodeIds = ids.slice(startIndex, endIndex)
+
+    const users = await pgdb.query(`
+      SELECT
+        u.*,
+        m."sequenceNumber" as "sequenceNumber"
+      FROM users u
+      JOIN memberships m
+        ON m.id = (SELECT id FROM memberships WHERE "userId" = u.id ORDER BY "sequenceNumber" ASC LIMIT 1) 
+      WHERE
+        ARRAY[u.id] && 
+        :ids;
+    `, {ids: nodeIds})
+
+    const endId = nodeIds[nodeIds.length - 1]
+    const startId = nodeIds[0]
+
+    return {
+      totalCount: ids.length,
+      nodes: users.map(transformUser),
+      pageInfo: {
+        endCursor: endId,
+        hasNextPage: endIndex < nodeIds.length - 1,
+        hasPreviousPage: startIndex > 0,
+        startCursor: startId
+      }
+    }
+  }
+
+  if (search) {
+    const userRows = await pgdb.query(`
+      SELECT
+        u.id
+      WHERE
+        u."isListed" = true AND u."isAdminUnlisted" = false AND
+        (
+          u."firstName" % :search OR
+          u."lastName" % :search OR
+          u."firstName" ILIKE :searchLike OR
+          u."lastName" ILIKE :searchLike OR
+          c.description % :search OR
+          c.description ILIKE :searchLike
+        );
```

Comment:
this is not a valid sql query 🧐

Reply:
totally didn't finish that one, done now: https://github.com/orbiting/republik-backend/pull/25/commits/4452e984fe59f97d7440bcc955f8da50f83988d1